public/js/components/Frames.js

```diff
@@ -66,6 +66,9 @@ const Frames = React.createClass({
 
   renderFrames() {
     let { frames } = this.props;
+    if (!frames) {
+      return null;
+    }
```

Comment:
Even though it's never going to get called unless `frames` is truthy, I think it's still good to structure functions so that they could individually pass a fuzz test. But how about:
```javascript
renderFrames(frames = []) {
    const numFramesToShow =
      this.state.showAllFrames ? frames.size : NUM_FRAMES_SHOWN;
    const framesToShow =
      frames.slice(0, numFramesToShow).map(frame => this.renderFrame(frame));

    return dom.ul({}, framesToShow);
  },
```

This makes the function's intent more clear — decouples it from `this.props` and annotates the type of `frames` by giving it a default. Even though that default is probably never going to get used, at least we've made it clear what `frames` is supposed to be.